diff --git a/cocoapods-patch-20220719-98210-9uxqla/LinkedinIOSHelper/LinkedInIOSHelper/LinkedInHelper/LinkedInAppSettings.m b/Pods/LinkedinIOSHelper/LinkedInIOSHelper/LinkedInHelper/LinkedInAppSettings.m
index 906eec3..63ae2fe 100644
--- a/cocoapods-patch-20220719-98210-9uxqla/LinkedinIOSHelper/LinkedInIOSHelper/LinkedInHelper/LinkedInAppSettings.m
+++ b/Pods/LinkedinIOSHelper/LinkedInIOSHelper/LinkedInHelper/LinkedInAppSettings.m
@@ -63,14 +63,14 @@
     self.grantedAccess = permissions;
     self.permissions = [NSMutableArray new];
 
-    if ([permissions containsObject:@(ContactInfo)])    [self.permissions addObject:@"r_contactinfo"];
+    if ([permissions containsObject:@(ContactInfo)])    [self.permissions addObject:@"r_liteprofile"];
     if ([permissions containsObject:@(FullProfile)])    [self.permissions addObject:@"r_fullprofile"];
     if (![permissions containsObject:@(FullProfile)] &&
         [permissions containsObject:@(BasicProfile)])   [self.permissions addObject:@"r_basicprofile"];
     if ([permissions containsObject:@(Nus)])            [self.permissions addObject:@"rw_nus"];
     if ([permissions containsObject:@(Network)])        [self.permissions addObject:@"r_network"];
     if ([permissions containsObject:@(EmailAddress)])   [self.permissions addObject:@"r_emailaddress"];
-    if ([permissions containsObject:@(Share)])          [self.permissions addObject:@"w_share"];
+    if ([permissions containsObject:@(Share)])          [self.permissions addObject:@"w_member_social"];
     if ([permissions containsObject:@(CompanyAdmin)])   [self.permissions addObject:@"rw_company_admin"];
     if ([permissions containsObject:@(Groups)])         [self.permissions addObject:@"rw_groups"];
     if ([permissions containsObject:@(Messages)])       [self.permissions addObject:@"w_messages"];
diff --git a/cocoapods-patch-20220719-98210-9uxqla/LinkedinIOSHelper/LinkedInIOSHelper/LinkedInHelper/LinkedInAuthorizationViewController.m b/Pods/LinkedinIOSHelper/LinkedInIOSHelper/LinkedInHelper/LinkedInAuthorizationViewController.m
index 73fbc14..498fac7 100644
--- a/cocoapods-patch-20220719-98210-9uxqla/LinkedinIOSHelper/LinkedInIOSHelper/LinkedInHelper/LinkedInAuthorizationViewController.m
+++ b/Pods/LinkedinIOSHelper/LinkedInIOSHelper/LinkedInHelper/LinkedInAuthorizationViewController.m
@@ -67,8 +67,10 @@ NSString * const linkedinIosHelperDomain = @"com.linkedinioshelper";
 - (void)viewDidAppear:(BOOL)animated {
     [super viewDidAppear:animated];
     
-    NSString *linkedIn = [NSString stringWithFormat:@"https://www.linkedin.com/uas/oauth2/authorization?response_type=code&client_id=%@&scope=%@&state=%@&redirect_uri=%@", self.serviceManager.settings.clientId, self.serviceManager.settings.grantedAccessString, self.serviceManager.settings.state, self.serviceManager.settings.applicationWithRedirectURL];
+    NSString *linkedIn = [NSString stringWithFormat:@"https://www.linkedin.com/oauth/v2/authorization?response_type=code&client_id=%@&scope=%@&state=%@&redirect_uri=%@", self.serviceManager.settings.clientId, self.serviceManager.settings.grantedAccessString, self.serviceManager.settings.state, self.serviceManager.settings.applicationWithRedirectURL];
     [self.webView loadRequest:[NSURLRequest requestWithURL:[NSURL URLWithString:linkedIn]]];
+    
+//https://www.linkedin.com/oauth/v2/authorization?response_type=code&client_id=86j2dodya1oazo&scope=r_liteprofile&state=foobar&redirect_uri=http://starupcode.com
 }
 
 - (void)viewWillLayoutSubviews {
diff --git a/cocoapods-patch-20220719-98210-9uxqla/LinkedinIOSHelper/LinkedInIOSHelper/LinkedInHelper/LinkedInServiceManager.m b/Pods/LinkedinIOSHelper/LinkedInIOSHelper/LinkedInHelper/LinkedInServiceManager.m
index 3717848..2c052d0 100644
--- a/cocoapods-patch-20220719-98210-9uxqla/LinkedinIOSHelper/LinkedInIOSHelper/LinkedInHelper/LinkedInServiceManager.m
+++ b/Pods/LinkedinIOSHelper/LinkedInIOSHelper/LinkedInHelper/LinkedInServiceManager.m
@@ -138,7 +138,7 @@
     
     __weak typeof (self) weakSelf = self;
     
-    LinkedInConnectionHandler *handler = [[LinkedInConnectionHandler alloc] initWithURL:[NSURL URLWithString:@"https://www.linkedin.com/uas/oauth2/accessToken"]
+    LinkedInConnectionHandler *handler = [[LinkedInConnectionHandler alloc] initWithURL:[NSURL URLWithString:@"https://www.linkedin.com/oauth/v2/accessToken"]
                                                                                    type:POST
                                                                                postData:postDataStr
                                                                                 success:^(NSDictionary *response) {
